src/Materials/effect.ts

```diff
@@ -612,6 +617,26 @@ export class Effect implements IDisposable {
         }
     }
 
+    private _getShaderCodeAndErrorLine(shader: WebGLShader, error: Nullable<string>, isFragment: boolean): [Nullable<string>, Nullable<string>] {
+        const regexp = isFragment ? /FRAGMENT SHADER ERROR: 0:(\d+?):/ : /VERTEX SHADER ERROR: 0:(\d+?):/;
+        const code = this._engine?._gl?.getShaderSource(shader) ?? null;

```

Comment:
Effect must not see gl (because for native or webgpu there is no gl). So we must come with a different way.

Probably exposing the getShaderSource on engine

Reply:
Ok, looking into that.

Reply:
I have added `_getShaderSource` on `ThinEngine` instead of `Engine` as it is in this class that the shaders are compiled, so it seemed appropriate(?)

Reply:
Yes!